### PR TITLE
[docs] Add redesigned markdown components

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Part of [ENG-3962](https://linear.app/expo/issue/ENG-3962/add-new-components-as-separate-prs-in-expoexpo)

This adds all generated markdown components, with the required margins/paddings. 

It does contain a small change in the generated component's display name. It's purely for debugging and understanding what is being wrapped inside what. Below you can see the screenshot of both without and with this displayname

without | with
--- | ---
without-displayname | with-displayname

# How

- Copied components from expo.dev and our docs fork
- Added explicit `displayName` to each component for improved debugging

# Test Plan

- Replace markdown components in `_app` to enable the new components
- See if all components are rendering properly

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/ui/components/Markdown/index.tsx
Comment (jonsamp) [inline docs/ui/components/Markdown/index.tsx:83]: We'll want this to be a component that renders code highlighting/syntax colors.

File: docs/ui/components/Markdown/index.tsx
Comment (jonsamp) [inline docs/ui/components/Markdown/index.tsx:29]: These headers feel too large to me. Here's a before (left) and after (right):

Screen Shot 2022-01-30 at 12 35 03 AM

One of the ways i made the design's content feel cleaner/better was by decreasing the sizes of the headers. For instance, a documentation title is an H3 (instead of an H1).

While we _can_ write components like `<H3 tag="h1">`, that doesn't seem right to me.

The typography I designed for the website is also used for marketing pages, which need to be bigger, whereas docs/long-form content should be smaller/more consistent. So! before we merge typography stuff, let me spend a little time working on setting some more typography. I think we need two sets, one for marketing and one for non-marketing pages.

Can we hold off on this PR so that I can set those styles and get them merged into /styleguide?

File: docs/ui/components/Markdown/index.tsx
Comment (byCedric) [reply docs/ui/components/Markdown/index.tsx:83]: I'll add a todo, but the new version isn't ready yet so not linked here.

File: docs/ui/components/Markdown/index.tsx
Comment (byCedric) [reply docs/ui/components/Markdown/index.tsx:29]: Yes! It's stacked on the typography PR, so we need to merge that before we can merge this. We could also try and wing it in the typography generation, where we still have `<H2>` (without `tag=h3`), but it has the H3 styling instead of H2?

I think I actually did that here too: https://github.com/byCedric/expo/blob/docs-redesign/docs/ui/components/Text/index.tsx#L54

**Comment by byCedric** [comment]
I'll hold off rebasing to soft-block this PR until we get the typography right.

File: docs/ui/components/Markdown/index.tsx
Comment (byCedric) [reply docs/ui/components/Markdown/index.tsx:29]: Solved with the new header sizes in styleguide

**Review by jonsamp — APPROVED** [review]

**Pull request merged**
